### ISSUE TITLE
Please consider improving how members are generated from "Implement Interface"

### ISSUE THREAD

**Issue by canton7** [dotnet/roslyn#53419]
This request is in two parts.

First, the case of auto-generating properties, by hitting the lightbulb over an interface which hasn't yet been implemented in a type and selecting "Implement Interface" e.g.

```cs
public interface IFoo
{
    int Id { get; set; }
}

public class Foo : [[IFoo]]
{
}
```

I have a memory that, once upon a time, this would generate:

```cs
public class Foo : IFoo
{
    public int Id { get; set; }
}
```

i.e. it the generated property would be an auto-prop which has the same accessors as the interface. This is very useful: 99% of my properties are auto-props, and it's easy to go and add explicit accessors if I need to.

These days (VS 16.8.3) however, it generates:

```cs
public class Foo : IFoo
{
    public int Id { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
}
```

This is just annoying -- I *always* need to go and edit this, by selecting multiple regions of code with my mouse, and deleting them.

Please consider going (back to?) the first behaviour, where the code generation assumes you want auto-props.

---------------

The second is the case of auto-generated method. Once upon a time, doing the same thing with:

```cs
public interface IFoo
{
    void Bar();
}
```

would generate:

```cs
public class Foo : IFoo
{
    public class Bar()
    {
        throw new NotImplementedException();
    }
}
```

This was great to work with -- I could plop my cursor into the method box, hit Ctrl-X to get rid of the exception, and start typing.

These days, however, I have `csharp_style_expression_bodied_methods = true:silent` in my .editorconfig, so this generates:

```cs
public class Foo : IFoo
{
    public class Bar() => throw new NotImplementedException();
}
```

Here, I need to select the `=> throw new NotImplementedException();`, delete, and type the `{` and `}` myself, before I can start typing.

I get that it's just following my .editorconfig preferences, but in this case I would prefer that it didn't, and always generate a full-bodied method. Most of my methods aren't single-line, and even in the case where I do want to replace this with a single-line method, it's easier to throw my cursor into the method body, Ctrl-X to get rid of the exception, type, then Ctrl+. to convert to an expression-bodied method. Alternatively, selecting the whole method body to delete it is easier than selecting the part of the line containing `=> throw new NotImplementedException();`

--------

Between these two niggles, it's at the point where implementing an interface by manually typing out each member is quicker than using the "Implement interface" lightbulb action, which is a sad.

Thanks!

**Comment by Youssef1313** [comment]
For the first part, this is already configurable:

![image](https://user-images.githubusercontent.com/31348972/118263130-29025b80-b4b6-11eb-9d19-65ab0077610c.png)

**Comment by canton7** [comment]
Ooh, genius. Thank you!

**Comment by Youssef1313** [comment]
@canton7 I'll defer to @CyrusNajmabadi since he implemented this feature in #18362.

**Comment by CyrusNajmabadi** [comment]
I think it's fine for the default to stay the way it is. There's been little feedback about this over the lifetime of the product.
